apps/builder/app/builder/features/style-panel/style-panel.tsx

```diff
@@ -10,35 +10,60 @@ import type { Publish } from "~/shared/pubsub";
 import { selectedBreakpointStore } from "~/shared/nano-states";
 import { useStyleData } from "./shared/use-style-data";
 import { StyleSettings } from "./style-settings";
-import { useCanvasWidth } from "~/builder/shared/nano-states";
+import { canvasWidthContainer } from "~/builder/shared/nano-states";
 import { StyleSourcesSection } from "./style-source-section";
 import { matchMedia } from "@webstudio-is/css-engine";
+import { useEffect, useState } from "react";
 
 type StylePanelProps = {
   publish: Publish;
   selectedInstance: Instance;
 };
 
+const useMatchMedia = () => {
+  const [matched, setMatched] = useState(false);
+
+  useEffect(() => {
+    return canvasWidthContainer.subscribe((canvasWidth) => {
+      const breakpoint = selectedBreakpointStore.get();
+      if (breakpoint === undefined) {
+        return;
+      }
+      if (canvasWidth === undefined) {
+        return;
+      }
+
+      if (matchMedia(breakpoint, canvasWidth) === false) {
+        setMatched(true);
+      }
+
+      setMatched(false);
```

Comment:
Do you intentionally set false right after true?

Reply:
no